tests/v2/validation/prime/prime_test.go

```diff
@@ -4,59 +4,89 @@ import (
 	"testing"
 
 	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	prime "github.com/rancher/rancher/tests/framework/extensions/prime"
 	"github.com/rancher/rancher/tests/framework/extensions/rancherversion"
 	"github.com/rancher/rancher/tests/framework/pkg/config"
 	"github.com/rancher/rancher/tests/framework/pkg/session"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 )
 
-const rancherBrand = "suse"
+const (
+	systemRegistry = "system-default-registry"
+	localCluster   = "local"
+	rancherBrand   = "suse"
+	uiBrand        = "ui-brand"
+)
 
-type PrimeVersionTestSuite struct {
+type PrimeTestSuite struct {
 	suite.Suite
 	session        *session.Session
 	client         *rancher.Client
 	isPrime        bool
 	rancherVersion string
+	primeRegistry  string
 }
 
-func (t *PrimeVersionTestSuite) TearDownSuite() {
+func (t *PrimeTestSuite) TearDownSuite() {
 	t.session.Cleanup()
 }
 
-func (t *PrimeVersionTestSuite) SetupSuite() {
+func (t *PrimeTestSuite) SetupSuite() {
 	testSession := session.NewSession()
 	t.session = testSession
-	primeConfig := new(rancherversion.Config)
 
+	primeConfig := new(rancherversion.Config)
 	config.LoadConfig(rancherversion.ConfigurationFileKey, primeConfig)
+
 	t.isPrime = primeConfig.IsPrime
 	t.rancherVersion = primeConfig.RancherVersion
+	t.primeRegistry = primeConfig.Registry
 
 	client, err := rancher.NewClient("", t.session)
 	assert.NoError(t.T(), err)
+
 	t.client = client
+}
+
+func (t *PrimeTestSuite) TestPrimeUIBrand() {
+	brand, err := t.client.Management.Setting.ByID(uiBrand)
+	require.NoError(t.T(), err)
 
+	checkBrand := prime.CheckUIBrand(t.client, t.isPrime, brand, rancherBrand)
+	assert.Equal(t.T(), rancherBrand, checkBrand)
```

Comment:
Like here what happens if the condition fails?

Reply:
Added error messages based on new conditions. Not sure if these new messages satisfy what you're looking for? There's error messaging but if the condition fails, the error trace just shows, but wouldn't that be fine to have since this is a focused test for Prime?